common/djangoapps/util/enterprise_helpers.py

```diff
@@ -225,3 +236,63 @@ def get_enterprise_branding_filter_param(request):
 
     """
     return request.session.get(ENTERPRISE_CUSTOMER_BRANDING_OVERRIDE_DETAILS, None)
+
+
+def get_dashboard_consent_notification(request, user, course_enrollments):
+    """
+    If relevant to the request at hand, create a banner on the dashboard indicating consent failed.
+    """
+    enrollment = None
+    enterprise_enrollment = None
+    course_id = request.GET.get(CONSENT_FAILED_PARAM)
+
+    if course_id:
+        enrollments = filter(
+            lambda enrollment: str(enrollment.course_id) == course_id,
+            course_enrollments
+        )
+        if enrollments:
+            enrollment = enrollments[0]
+
+        try:
+            enterprise_enrollment = EnterpriseCourseEnrollment.objects.get(
+                course_id=course_id,
+                enterprise_customer_user__user_id=user.id,
+            )
+        except EnterpriseCourseEnrollment.DoesNotExist:
+            pass
+
+    if enterprise_enrollment and enrollment:
+        enterprise_customer = enterprise_enrollment.enterprise_customer_user.enterprise_customer
+        contact_info = getattr(enterprise_customer, 'contact_email', None)
+
+        if contact_info is None:
+            message_template = _(
+                'If you have concerns about sharing your data, please contact your '
+                'administrator at {enterprise_customer_name}.'
+            )
+        else:
+            message_template = _(
+                'If you have concerns about sharing your data, please contact your administrator '
```

Comment:
If you drop the `administrator` down to the next line it will be clearer to the reader that these two templates are the same aside from the "at {contact_info}" part.  I'm okay with the duplication in this case because having full sentences will be easier for translators to interpret.